### PR TITLE
#21 Remove all Native modules before move to Elm v0.19

### PR DESCRIPTION
**Pull request by SergKam**
No Native modules allowed in Elm v0.19
So before migration, we need to get rid of all the "Native" modules.
We could do it using "ports" but I want to experiment with the idea of using XHR request interceptor
to treat Bowser as a node.js server with expres.js like API for Elm (see /client/ElmExpress.js). 
```js
// index.js
const ElmExpress = require('./ElmExpress')
ElmExpress.create(window, router => {

      router.get('localStorage', (req, res) =>
          res.text(window.localStorage.getItem(req.params.get('key'))))

      router.post('localStorage', (req, res) =>
          res.ok(window.localStorage.setItem(req.params.get('key'), req.body)))
})
```

It is very easy to use. Any request to `elm:` will be intercepted by the browser. 
```elm
fetchAll : (Result Error String -> Msg)  -> Cmd Msg
fetchAll tagger =
    Http.send tagger <|
        Http.getString "elm:localStorage?key=starred.eventTypes"


saveAll : (Result Error String -> Msg)  -> String -> Cmd Msg
saveAll tagger data =
    Http.send tagger <|
        Http.post "elm:localStorage?key=starred.eventTypes" (Http.stringBody "" data) string

```
- Native Ace rich editor replaced by Web Component (custom element)
- Request for element "width" replaced by parsing event.target.clientWidth
- internalLink moved to the `Routing.Helpers` module to break circular dependancy.
- `link` replace by `internalLink`   
- I use the new version of elm-format. Unfortunately, it produces different formatting, resulting in too many lines changed.

### CODE REVIEW

**Comment by SergKam** [comment]
:+1:

File: client/Helpers/UI.elm [hunk 1]
```diff
@@ -113,11 +112,15 @@ grid columns rows =
 
 {-| Create 'onselect' event listener attribute for select HTML node
 -}
-onSelect : (String -> msg) -> Html.Attribute msg
-onSelect msg =
+onChange : (String -> msg) -> Html.Attribute msg
```
Comment (lmontrieux) [inline client/Helpers/UI.elm]: This now differs from the documentation line above

File: client/js/aceEditor.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    session.setUseSoftTabs(this.getAttribute('softtabs'))
+    this.getAttribute('tabsize') && session.setTabSize(+this.getAttribute('tabsize'))
+    session.setUseWrapMode(this.hasAttribute('wrapmode'))
+
+
+    // Observe input textNode changes
+    // Could be buggy as editor was also added to Light DOM;
+    const observer = new MutationObserver(function (mutations) {
+      mutations.forEach(function (mutation) {
+        // console.log("observation", mutation.type, arguments, mutations, editor, text);
```
Comment (lmontrieux) [inline client/js/aceEditor.js]: Should this line go away?

File: client/js/aceEditor.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+  connectedCallback() {
+    const text = this.childNodes[0]
+    const container = this.container
+    const element = this
+    let editor
+
+    if (this.editor) {
+      editor = this.editor
+    } else {
+      // container.appendChild(text);
```
Comment (lmontrieux) [inline client/js/aceEditor.js]: Should this line go away?

File: client/js/aceEditor.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    const observer = new MutationObserver(function (mutations) {
+      mutations.forEach(function (mutation) {
+        // console.log("observation", mutation.type, arguments, mutations, editor, text);
+        if (mutation.type === 'characterData') {
+          element.value = text.data
+        }
+      })
+    })
+    text && observer.observe(text, {characterData: true})
+    // container.appendChild(text);
```
Comment (lmontrieux) [inline client/js/aceEditor.js]: Should this line go away?

File: client/Helpers/UI.elm (hunk 1 above)
Comment (SergKam) [reply client/Helpers/UI.elm]: `onSelect` is redundant now. removed

File: client/assets/static/worker-json.js [hunk 5]
```diff
@@ -0,0 +1 @@
... t}),Object.isSealed||(Object.isSealed=function(t){return!1}),Object.isFrozen||(Object.isFrozen=function(t){return!1}),Object.isExtensible||(Object.isExtensible=function(t){if(Object(t)===t)throw new TypeError;var n="";while(f(t,n))n+="?";t[n]=!0;var r=f(t,n);return delete t[n],r});if(!Object.keys){var L=!0,A=["toString","toLocaleString","valueOf","hasOwnProperty","isPrototypeOf","propertyIsEnumerable","constructor"],O=A.length;for(var M in{toString:null})L=!1;Object.keys=function I(e){if(typeof e!="object"&&typeof e!="function"||e===null)throw new TypeError("Object.keys called on a non-object");var I=[];for(var t in e)f(e,t)&&I.push(t);if(L)for(var n=0,r=O;n<r;n++){var i=A[n];f(e,i)&&I.push(i)}return I}}Date.now||(Date.now=function(){return(new Date).getTime()});var _="	\n\x0b\f\r \u00a0\u1680\u180e\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u202f\u205f\u3000\u2028\u2029\ufeff";if(!String.prototype.trim||_.trim()){_="["+_+"]";var D=new RegExp("^"+_+_+"*"),P=new RegExp(_+_+"*$");String.prototype.trim=function(){return String(this).replace(D,"").replace(P,"")}}var F=function(e){if(e==null)throw new TypeError("can't convert "+e+" to object");return Object(e)}})
```
Comment (lmontrieux) [inline client/assets/static/worker-json.js]: Is this auto-generated?

Comment (SergKam) [reply client/assets/static/worker-json.js]: No, it is part of Ace editor package.
The origin of it is described here.
https://github.com/zalando-nakadi/nakadi-ui/pull/51/files#diff-470139a8fcf22ed0cc1e28228af6ae73R21

**Comment by lmontrieux** [comment]
👍

**Comment by SergKam** [comment]
:+1:

**Pull request merged**